src/main/kotlin/org/rust/ide/annotator/fixes/MakePublicFix.kt

```diff
@@ -70,16 +70,15 @@ class MakePublicFix(
     }
 
     companion object {
-        fun createIfCompatible(visible: RsVisible, elementName: String?, withinOneCrate: Boolean): MakePublicFix? {
-            return when {
-                // TODO: Allow this fix for pub-restricted elements too
-                visible.visibility is RsVisibility.Private
-                    && visible !is RsEnumVariant
-                    && visible.parent.parent !is RsTraitItem
-                    && visible.containingCrate?.origin == PackageOrigin.WORKSPACE
-                -> MakePublicFix(visible, elementName, withinOneCrate)
-                else -> null
-            }
+        fun createIfCompatible(
+            visible: RsVisibilityOwner,
+            elementName: String?,
+            withinOneCrate: Boolean
+        ): MakePublicFix? {
+            if (visible is RsEnumVariant ||
```

Comment:
Maybe it's worth to return `null` in case `visible.visibility is RsVisibility.Public` as well

Reply:
I don't think it's really needed here.
In my opinion, `visible.visibility is RsVisibility.Public` it's a part of a condition that inspection/annotator should check to provide some annotation at all (even without quick-fix) so the addition of this code here will be just a duplication of inspection/annotator code.
Other conditions in `createIfCompatible ` are more about quick fix itself and its current restrictions of implementation